game-app/game-core/src/main/java/games/strategy/triplea/ui/UnitChooser.java

```diff
@@ -648,12 +683,183 @@ public Dimension getMinimumSize() {
+
+    private UiContextDecoration(final UiContext uiContext) {
+      this.uiContext = uiContext;
+      final var unitImageFactory = uiContext.getUnitImageFactory();
+      final int unitImageHeight = unitImageFactory.getUnitImageHeight();
+      final double nonWithdrawableImageHeight = getNonWithdrawableImageHeight(unitImageHeight);
+      nonWithdrawableImage = loadNonWidthdrawableImage((int) nonWithdrawableImageHeight);
+
+      final double scaleChange =
+          (double) nonWithdrawableImage.getHeight() / nonWithdrawableImageHeight;
+      // scaleChange is the factor by which the scale factor of this.unitImageFactory
+      // must be larger than the scale factor of uiContext.getUnitImageFactory()
+      // so the non-withdrawable image has the right size in comparison to the unit icon.
+
+      final double scaleFactor = scaleChange * unitImageFactory.getScaleFactor();
+      // With this scaleFactor the unit images will have the correct height in relation
+      // to the non-withdrawable image.
+      // The relation is controlled by getNonWithdrawableImageHeight. By the time of writing
+      // this, getNonWithdrawableImageHeight makes the non-withdrawable image half as high as
+      // the unit images.
+      // At the time of writing this, there are two variants of the non-whithdrawable image
+      // being 24 pixels resp. 32 pixels high.
+      // So the unit image will be either 48 pixels or 32 pixels high.
+
+      this.unitImageFactory = unitImageFactory.withScaleFactor(scaleFactor);
+
+      Postconditions.assertState(nonWithdrawableImage.getHeight()
+          == getNonWithdrawableImageHeight(this.unitImageFactory.getUnitImageHeight()));
+    }
+
+    /**
+     * @param nonWithdrawableImageHeight height the non-withdrawable image
+     *  would have in the came - but not on the <code>UnitChooser</code>,
+     *  because on the <code>UnitChooser</code> a specific scale is being used to make sure
+     *  that the non-withdrawable image can reasonably be displayed with its original height,
+     *  i.e. unscaled.
+     *
+     * @return the non-withdrawable image best matching the unit image height
+     *  given the proportion between non-withdrawable imagehight and unit image height
+     *  determined by <code>getNonWithdrawableImageHeight</code>.
+     */
+
+    private static BufferedImage loadNonWidthdrawableImage(final int nonWithdrawableImageHeight) {
+      final BufferedImage imgSmall = loadGenericUnitsImage("non-withdrawable_small.png");
+
+      if (nonWithdrawableImageHeight <= imgSmall.getHeight(null)) {
+        return imgSmall;
+      } else {
+        final BufferedImage imgBig = loadGenericUnitsImage("non-withdrawable.png");
+
+        return nonWithdrawableImageHeight < imgBig.getHeight(null) ? imgSmall : imgBig;
+      }
+    }
+
+    private static BufferedImage loadGenericUnitsImage(final String fileName) {
+      return EngineImageLoader.loadImage("units","generic", fileName);
```

Comment:
This like should be "misc" instead of "units", "generic"

Reply:
I had the images in `misc` before. You wanted me to move them to `units`.

Reply:
Sorry for any conflicting recommendations. We do need the path here to agree with the path in `assets`.